docs/extra_features.rst

```diff
@@ -0,0 +1,69 @@
+Extra Features
+==============
+
+July 2, 2021
+
+SpiceyPy was always intended to be a thin wrapper around the CSPICE routines, however
+as it is implemented in Python there are a small number of extra features to help
+make end-users' code more Pythonic.
+
+Overview
```

Comment:
So I actually think it would clearer if this documentation can follow more of a narrative approach that I have in other areas like https://spiceypy.readthedocs.io/en/main/exceptions.html#not-found-errors, and for it to not mirror the spice lessons layout. 

currently all the important details are inside the python code block. I think the current examples are fine, just break them apart into separate code blocks, change in in-line comments to paragraphs outside the code blocks, and remove the function def for the first example. 

The task statement/ learning goals can then be removed.